gfx/drivers_shader/slang_process.cpp

```diff
@@ -0,0 +1,406 @@
+
+#include <fstream>
+#include <iostream>
+#include <spirv_hlsl.hpp>
+#include <stdint.h>
+
+#include "verbosity.h"
+#include "glslang_util.hpp"
+#include "slang_preprocess.h"
+#include "slang_reflection.h"
+#include "slang_process.h"
+
+using namespace spirv_cross;
+using namespace std;
+
+template <typename P>
+static bool set_unique_map(unordered_map<string, P>& m, const string& name, const P& p)
+{
+   auto itr = m.find(name);
+   if (itr != end(m))
+   {
+      RARCH_ERR("[slang]: Alias \"%s\" already exists.\n", name.c_str());
+      return false;
+   }
+
+   m[name] = p;
+   return true;
+}
+
+template <typename M, typename S>
+static string get_semantic_name(const unordered_map<string, M>* map, S semantic, unsigned index)
+{
+   for (const pair<string, M>& m : *map)
+   {
+      if (m.second.semantic == semantic && m.second.index == index)
+         return m.first;
+   }
+   return string();
+}
+
+static string
+get_semantic_name(slang_reflection& reflection, slang_semantic semantic, unsigned index)
+{
+   static const char* names[] = {
+      "MVP",
+      "OutputSize",
+      "FinalViewportSize",
+      "FrameCount",
+   };
+   if ((int)semantic < sizeof(names) / sizeof(*names))
+      return std::string(names[semantic]);
+
+   return get_semantic_name(reflection.semantic_map, semantic, index);
+}
+
+static string
+get_semantic_name(slang_reflection& reflection, slang_texture_semantic semantic, unsigned index)
+{
+   static const char* names[] = {
+      "Original", "Source", "OriginalHistory", "PassOutput", "PassFeedback",
+   };
+   if ((int)semantic < (int)SLANG_TEXTURE_SEMANTIC_ORIGINAL_HISTORY)
+      return std::string(names[semantic]);
+   else if ((int)semantic < sizeof(names) / sizeof(*names))
+      return std::string(names[semantic]) + to_string(index);
+
+   return get_semantic_name(reflection.texture_semantic_map, semantic, index);
+}
+
+static string get_size_semantic_name(
+      slang_reflection& reflection, slang_texture_semantic semantic, unsigned index)
+{
+   static const char* names[] = {
+      "OriginalSize", "SourceSize", "OriginalHistorySize", "PassOutputSize", "PassFeedbackSize",
+   };
+   if ((int)semantic < (int)SLANG_TEXTURE_SEMANTIC_ORIGINAL_HISTORY)
+      return std::string(names[semantic]);
+   if ((int)semantic < sizeof(names) / sizeof(*names))
+      return std::string(names[semantic]) + to_string(index);
+
+   return get_semantic_name(reflection.texture_semantic_uniform_map, semantic, index);
+}
+
+static bool slang_process_reflection(
```

Comment:
Is this file completely new? Some of this stuff seems familiar ...

Reply:
yeah it is new, I did copy some parts from vulkan_shaders and slang_reflection though, that's why it seems familiar.

Reply:
Will Vulkan use the new file then? I assume it's a superset of the old functionality to support HLSL.